timemachine/potentials/nonbonded.py

```diff
@@ -131,6 +132,51 @@ def convert_exclusions_to_rescale_masks(exclusion_idxs, scales, N):
     return charge_rescale_mask, lj_rescale_mask
 
 
+def filter_exclusions(
+    atom_idxs: NDArray[np.int32],
+    exclusion_idxs: NDArray[np.int32],
+    scale_factors: NDArray[np.float64],
+    update_idxs=False,
+):
+    """
+    Return the exclusions and corresponding scale factors
+    with the atoms not in atom_idxs removed.
+
+    Parameters
+    ----------
+    atom_idxs:
+        List of atoms that are considered for interaction.
+
+    exclusion_idxs:
+        List of atom pairs to exclude.
+
+    scale_factors:
+        Per exclusion charge and lj scale factors.
+
+    update_idxs:
+        Set to True to remap the exclusion indexes
+        to point to the index of atom_idxs.
+        This can be used for the reference JAX potential.
+    """
+    atom_idxs_set = set(atom_idxs)
+    map_to_filtered = {j: i for i, j in enumerate(atom_idxs)}
+    filtered_exclusion_idxs_ = []
+    filtered_scale_factors_ = []
+    for (i, j), sf in zip(exclusion_idxs, scale_factors):
+        if i not in atom_idxs_set or j not in atom_idxs_set:
+            continue
+        if update_idxs:
+            i, j = map_to_filtered[i], map_to_filtered[j]
```

Comment:
Can you explain this line a bit more? Since these exclusions go into a NonbondedPairList, shouldn't the indices stay the same, and only the filtering matter?

Reply:
Nevermind, had misread the code

Reply:
That is correct for the GPU implementation.

In the reference jax implementation, the conformations/parameters are filtered by remapping (removing) the idx not in atom_idxs. I.e. `conf = jnp.array(conf)[atom_idxs, :]`